services/publicapi/test/public_api_test.go

```diff
@@ -86,6 +72,64 @@ func TestSendTransaction_BlocksUntilTransactionCompletes(t *testing.T) {
 	})
 }
 
+func TestSendTransaction_BlocksUntilTransactionErrors(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		harness := newPublicApiHarness(ctx, 1*time.Second)
+
+		blockTime := primitives.TimestampNano(time.Now().Nanosecond())
+		txb := builders.Transaction().Builder()
+		txHash := digest.CalcTxHash(txb.Build().Transaction())
+
+		harness.txpMock.When("AddNewTransaction", mock.Any).Times(1).
```

Comment:
you can move the mock setup into the harness, so that the test only needs to supply the callback function. Everything else, including the sleep (yuck) can be hidden in the harness:
```golang
harness.OnAddNewTransaction(func() {
  harness.papi.HandleTransactionError(...)
})
```